src/WebAuthn.sol

```diff
@@ -122,47 +109,41 @@ library WebAuthn {
             return false;
         }
 
-        // 11. and 12. will be verified by the signature check
-        // 11. Verify that the value of C.type is the string webauthn.get.
+        string memory _type = webAuthnAuth.clientDataJSON.slice(webAuthnAuth.typeIndex, webAuthnAuth.typeIndex + 21);
```

Comment:
@mdehoog @rholterhus @xenoliss given all the steps we skip from the official spec, I am wondering if we care to keep this. Do we *really* care if it is `webauthn.get` vs. `.create` not sure.

Reply:
Two questions:
 - what's the worse that could happen if we skip?
 - how much gas do we save by skipping (seems negligible at first glance)?

Reply:
For gas saving, I won't comment as I haven't tested it.

> What's the worst that could happen if we skip?

The [specification](https://w3c.github.io/webauthn/#dom-collectedclientdata-type) is quite clear about it.

I guess whether to skip it or not depends on whether we want to restrict its usage to our use case only or make it more generic:


- For our usage only, I think skipping it is ok. In our specific case, we are not really interested in its value and simply assume that if the user signed it, then they must agree with whatever they just signed.

- For public usage (or other usages not related to our ERC4337 implementation), it might make sense to keep this verification to avoid "certain types of signature confusion attacks."

I'm always in favor of security, so if the gas is negligible, I would keep it. If it breaks at some point because an authenticator delivers assertion signatures with a type different than "webauthn.get", we can still upgrade the contract.

Reply:
My opinion would also be to keep the check. I agree with the above analysis - the worst-case scenario would be if someone could reuse a signature from a registration ceremony. For the context of ERC4337, this is very unlikely to matter. Because the `challenge` in the registration ceremony has no reason to be equal to a hash of a UserOp, so there shouldn't ever be a collision. But for public usage it'd be good to keep the check.

Reply:
Yeah I guess the only thought I had was whether we wanted to allow `webauthn.create` to be used for signing. But I am ok with the conclusion here. Will keep.